dash/development/base_component.py

```diff
@@ -387,6 +387,120 @@ def __repr__(self):
+    # it to be `null` or whether that was just the default value.
+    # The solution might be to deal with default values better although
+    # not all component authors will supply those.
+    c = '''{docstring}
+
+    html{typename} <- function(..., {default_argtext}) {{
+
+    component <- list(
+      props = list(
+         {default_paramtext}
+      ),
+      type = '{typename}',
+      namespace = '{namespace}',
+      propNames = c({list_of_valid_keys}),
+      package = '{package_name}'
+    )
+
+    component$props <- filter_null(component$props)
+    component <- append_wildcard_props(component, wildcards = c({default_wildcards}), ...)
+
+    structure(component, class = c('dash_component', 'list'))
+    }}
+'''
+
+    filtered_props = reorder_props(filter_props(props))
+    # pylint: disable=unused-variable
+    list_of_valid_wildcard_attr_prefixes = repr(parse_wildcards(props))
+    # pylint: disable=unused-variable
+    list_of_valid_keys = repr(list(map(str, filtered_props.keys())))
+    list_of_valid_keys = list_of_valid_keys[1:-1]
+    package_name = make_package_name(namespace)
+    # pylint: disable=unused-variable
+    docstring = create_docstring_r(
+        component_name=typename,
+        props=filtered_props,
+        events=parse_events(props),
+        description=description).replace('\r\n', '\n')
+
+    # pylint: disable=unused-variable
+    # want to remove data-*, aria-*
+    events = '[' + ', '.join(parse_events(props)) + ']'
+    prop_keys = list(props.keys())
+    default_paramtext = ''
+    default_wildcards = ''
+    wildcard_list = ''
+    for key in props:
+        if '*' in key:
+            wildcard_list.join(key)
+    if 'children' in props:
+        prop_keys.remove('children')
+        default_argtext = "children=NULL, "
+        # pylint: disable=unused-variable
+        argtext = 'children=children, **args'
+    else:
+        default_argtext = ""
+        argtext = '**args'
+    # in R, we set parameters with no defaults to NULL, here we'll do that if no default value exists
+    default_wildcards += ", ".join(
+        [('\'{:s}\''.format(p))
+         for p in prop_keys
+         if '*' in p]
+    )
+    default_argtext += ", ".join(
+        [('{:s}={}'.format(p, props[p]['defaultValue']['value'])
+          if 'defaultValue' in props[p] else
+          '{:s}=NULL'.format(p))
+         for p in prop_keys
+         if not p.endswith("-*") and
+         p not in keyword.kwlist and
+         p not in ['setProps']]
+    )
+    default_paramtext += ", ".join(
+            [('{:s}={:s}'.format(p, p)
+               if p != "children" else
+               '{:s}=c(children, assert_valid_children(..., wildcards = c({:s})))'.format(p, default_wildcards))
+             for p in props.keys()
+             if not p.endswith("-*") and
+             p not in keyword.kwlist and
+             p not in ['setProps']]
+    )
+    required_args = required_props(props)
+    return c.format(**locals())
```

Comment:
I'd like them to be explicitly stated instead of locals(), then we can remove all those `# pylint: disable=unused-variable` and easily see what variables are not used.

Reply:
Right, I think we talked about this when you explained list comprehensions to me a while ago. Will remove the shorthand and explicitly declare the variables.

Reply:
I agree with this, and we should do it in the Python version as well. I've added a note here: https://github.com/plotly/dash/issues/311